### PR TITLE
Add optional PostImportScriptPath parameter to lazy-posh-git module to enable running custom script after posh-git import

### LINKED ISSUE #1: How to provide a way to set custom `$gitpromptsettings`
I'm enhusiastic about lazy-posh-git. Simple, effective, nice!

My only issue so far: I can't set custom settings for posh-git in my profile anymore.

```powershell
import-module lazy-posh-git
$GitPromptSettings.DefaultPromptAbbreviateHomeDirectory = $true
$GitPromptSettings.AfterStatus.Text = "]`n"
```
Ofcourse fails with:
```text
InvalidOperation: C:\Users\user\Documents\PowerShell\profile.ps1:2
Line |
   2 |  $GitPromptSettings.DefaultPromptAbbreviateHomeDirectory = $true
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The property 'DefaultPromptAbbreviateHomeDirectory' cannot be found on this object. Verify that the property exists and can be set.

InvalidOperation: C:\Users\user\Documents\PowerShell\profile.ps1:3
Line |
   3 |  $GitPromptSettings.AfterPath = "]`n"
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The property 'AfterPath' cannot be found on this object. Verify that the property exists and can be set.
```

This makes sense: the variable `$GitPromptSettings` is not created yet, since we are deferring the import of the posh-git module until a later moment in time. 

Related issue: https://github.com/dahlbyk/posh-git/issues/635

Since `posh-git` does not support a settings file at this time, I thought about adding delayed settings load to `lazy-posh-git`. There could be various ways of doing this. 

One way would be to support a settings script that, if it exists, could be loaded after importing `posh-git`:

```powershell
...
        if (Test-Path (Join-Path $PWD.Path '.git')) {
            Import-Module posh-git
            if (Test-Path (Join-Path $PsScriptroot settings.ps1)) {
                . (Join-Path $PsScriptroot settings.ps1)
            }
        }
...
```
Where there can be a `settings.ps1` file in the root of the `lazy-posh-git` module with a settingsfile like this:
```powershell
$GitPromptSettings.DefaultPromptAbbreviateHomeDirectory = $true
$GitPromptSettings.AfterStatus.Text = "]`n"
```

I'm curious about your thoughts on this.

### PR DESCRIPTION
**Pull request by Jawz84**
Fixes #1

This PR is intended to come up with the best solution to enable lazy-posh-git to load additional settings, after posh-git is imported.

Please feel free to make improvements, or suggestions.

### CODE REVIEW

File: lazy-posh-git.psm1 [hunk 1]
```diff
@@ -2,6 +2,10 @@
     .ForwardHelpTargetName Microsoft.PowerShell.Management\Set-Location
     .ForwardHelpCategory Cmdlet
 #>
+param(
+    [string] $PoshGitSettingsScriptPath
```
Comment (bergmeister) [inline lazy-posh-git.psm1]:
```suggestion
    [Parameter(Position = 0)]
    [string] $PostImportScriptPath
```

File: lazy-posh-git.psm1 [hunk 2]
```diff
@@ -81,6 +85,9 @@ function Set-Location {
         }
         if (Test-Path (Join-Path $PWD.Path '.git')) {
             Import-Module posh-git
+            if (Test-Path $PoshGitSettingsScriptPath) {
```
Comment (bergmeister) [inline lazy-posh-git.psm1]:
```suggestion
            if (Test-Path $PostImportScriptPath) {
```

File: lazy-posh-git.psm1 [hunk 3]
```diff
@@ -81,6 +85,9 @@ function Set-Location {
...
+                & $PoshGitSettingsScriptPath
```
Comment (bergmeister) [inline lazy-posh-git.psm1]:
```suggestion
                & $PostImportScriptPath
```

File: README.md [hunk 4]
```diff
@@ -20,6 +20,20 @@ Import-Module 'lazy-posh-git'
 
 Only once you navigate to a git directory, the `posh-git` module will be imported, which is when it is needed, therefore defering the delay caused by the import to the last moment in time until we know we need it.
 
+## Using custom posh-git settings
+If there are certain posh-git settings you wish to change after posh-git is loaded, you can do so. Create a script file with the desired settings in a convenient location. Then tell lazy-posh-git where to find it, like this:
```
Comment (bergmeister) [inline README.md]:
```suggestion

If there are certain `posh-git` settings you wish to change after `posh-git` is imported or running a custom script after import, you can do so by passing the path to a script as the first argument to the `PoshGitSettingsScriptPath` parameter when importing `lazy-posh-git`:
```

File: README.md [hunk 5]
```diff
@@ -20,6 +20,20 @@ Import-Module 'lazy-posh-git'
 
 Only once you navigate to a git directory, the `posh-git` module will be imported, which is when it is needed, therefore defering the delay caused by the import to the last moment in time until we know we need it.
 
+## Using custom posh-git settings
```
Comment (bergmeister) [inline README.md]:
```suggestion
## Using custom posh-git settings or running a custom script after import
```

File: README.md [hunk 6]
```diff
@@ -20,6 +20,20 @@ Import-Module 'lazy-posh-git'
...
+If there are certain posh-git settings you wish to change after posh-git is loaded, you can do so. Create a script file with the desired settings in a convenient location. Then tell lazy-posh-git where to find it, like this:
+
+```powershell
+Import-Module 'lazy-posh-git' -ArgumentList '~\path-to-my\git-prompt-settings.ps1'
+``` 
+
+Lazy-posh-git will run your script directly after it has loaded posh-git. Your posh-git settings script could look like this for example:
```
Comment (bergmeister) [inline README.md]:
```suggestion
`Lazy-posh-git` will run this script immediately after it has imported `posh-git`. The path can be absolute or relative depending on the assumptions you can make. Your script could look like this for example:
```

**Review by bergmeister — APPROVED** [review]
Looks good, thanks, tested as well. Just added suggestions to rename parameter to be more generic since one could run any custom script for whatever reason, add parameter position and tweaks to docs.

**Comment by Jawz84** [comment]
Yes, nice teamwork! I agree more generic naming is better here! LGTM 🚀

**Comment by bergmeister** [comment]
🎉v0.1.0 has been released which incorporates this pull request.🎉

Handy links:
- [Release Notes](https://github.com/bergmeister/lazy-posh-git/releases/tag/0.1.0)

**Pull request merged**